bin/deploy-vhost

```diff
@@ -133,140 +133,56 @@ sub get_pgpw_string {
 }
 
-sub one_db_conf_php {
-    my ($params, $database, $internal) = @_;
-    my $port_line       = '';
-    my $db_ro_host_line = '';
-    my $port_ro_line    = '';
-    if ($params->{type} eq 'psql') {
-        $port_line = "define('OPTION_$params->{prefix}_DB_PORT', $params->{port});";
-        if ($params->{replica}) {
-            $db_ro_host_line = "define('OPTION_$params->{prefix}_DB_RO_HOST', '$params->{replica}.ukcod.org.uk');";
-            # Hot standby usually on 5433, unless it's been failed over at some point.
-            if ($params->{port} eq '5432') {
-                $port_ro_line = "define('OPTION_$params->{prefix}_DB_RO_PORT', 5433);";
-            }
-            else {
-                $port_ro_line = "define('OPTION_$params->{prefix}_DB_RO_PORT', 5432);";
-            }
-        }
-    }
-    my $db_host = get_db_host($params, $internal);
-    my $pgpw_string = get_pgpw_string($params, $database);
-
-    my $conf = <<END;
-define('OPTION_$params->{prefix}_DB_HOST', '$db_host');
-$port_line
-$db_ro_host_line
-$port_ro_line
-define('OPTION_$params->{prefix}_DB_NAME', '$database');
-define('OPTION_$params->{prefix}_DB_USER', '$database');
-define('OPTION_$params->{prefix}_DB_PASS', '\${\\(pgpw('$pgpw_string'))}');
-END
-    return $conf;
-
-}
-
-sub one_db_conf_yml {
-    my ($params, $database, $internal) = @_;
-    my $port_line       = '';
-    my $db_ro_host_line = '';
-    my $port_ro_line    = '';
-    if ($params->{type} eq 'psql') {
-        $port_line = "$params->{prefix}_DB_PORT: $params->{port}";
-        if ($params->{replica}) {
-            $db_ro_host_line = "$params->{prefix}_DB_RO_HOST: '$params->{replica}.ukcod.org.uk'";
-            # Hot standby usually on 5433, unless it's been failed over at some point.
-            if ($params->{port} eq '5432') {
-                $port_ro_line = "$params->{prefix}_DB_RO_PORT: 5433";
-            }
-            else {
-                $port_ro_line = "$params->{prefix}_DB_RO_PORT: 5432";
-            }
-        }
-    }
-    my $db_host = get_db_host($params, $internal);
-    my $pgpw_string = get_pgpw_string($params, $database);
-
-    my $conf = <<END;
-$params->{prefix}_DB_HOST: '$db_host'
-$port_line
-$db_ro_host_line
-$port_ro_line
-$params->{prefix}_DB_NAME: '$database'
-$params->{prefix}_DB_USER: '$database'
-$params->{prefix}_DB_PASS: '\${\\(pgpw('$pgpw_string'))}'
-END
-    return $conf;
-
-}
 
+# For each database, pull in information from the JSON and derive some
+# further data as required. Store this in $conf->{database_configs}
 sub db_conf {
     my ($params, $database) = @_;
     if ($params->{type} ne 'mysql' && $params->{type} ne 'psql' && $params->{type} ne 'mongo'){
         die "unknown database type '$params->{type}'";
     }
 
-    # Strings for PHP configuration
-    $conf->{database_configs}{default} .= one_db_conf_php($params, $database, 1);
-    $conf->{database_configs}{external} .= one_db_conf_php($params, $database, 0);
```

Comment:
Doesn't really matter, but the removal of the external bits seems to me separate from the refactoring out of the code (indeed, you've done one last bit in another commit), it'd be clearer if they had been removed first on their own, then the code moved, iyswim.